### PR TITLE
ASoC: sdca: test adev before calling acpi_dev_for_each_child

### PR DESCRIPTION
**Pull request by bardliao**
sdca_lookup_functions may be called by a Peripheral that is not listed in the ACPI table. Testing adev is required to avoid kernel NULL pointer dereference.

### CODE REVIEW

File: sound/soc/sdca/sdca_functions.c [hunk 1]
```diff
@@ -165,6 +165,10 @@ void sdca_lookup_functions(struct sdw_slave *slave)
 	struct device *dev = &slave->dev;
 	struct acpi_device *adev = to_acpi_device_node(dev->fwnode);
 
+	if (!adev) {
+		dev_dbg(dev, "Not an ACPI device\n");
```
Comment (ujfalusi) [inline sound/soc/sdca/sdca_functions.c]: I would move this to info level to be visible and rephrase the print to something like 'No matching ACPI device found, ignoring peripheral`

**Review by ujfalusi — COMMENTED** [review]
@bardliao, nice catch!

File: sound/soc/sdca/sdca_functions.c (hunk 1 above)
Comment (bardliao) [reply sound/soc/sdca/sdca_functions.c]: Updated

**Review by ujfalusi — APPROVED** [review]

File: sound/soc/sdca/sdca_functions.c (hunk 1 above)
Comment (ujfalusi) [reply sound/soc/sdca/sdca_functions.c]: Btw, do you need/want Fixes tag?

**Review by lgirdwood — APPROVED** [review]

File: sound/soc/sdca/sdca_functions.c (hunk 1 above)
Comment (bardliao) [reply sound/soc/sdca/sdca_functions.c]: > Btw, do you need/want Fixes tag?

Yes, done.

**Review by ujfalusi — APPROVED** [review]

**Review by ranj063 — APPROVED** [review]

**Pull request merged**